trieste/models/keras/architectures.py

```diff
@@ -0,0 +1,315 @@
+    :param ensemble_size: The size of the ensemble, that is, the number of base learners or
+        individual neural networks in the ensemble.
+    :param num_hidden_layers: The number of hidden layers in each network.
+    :param units: The number of nodes in each hidden layer.
+    :param activation: The activation function in each hidden layer.
+    :param independent: If set to `True` then :class:`~tfp.layers.IndependentNormal` layer
+        is used as the output layer. This models outputs as independent, only the diagonal
+        elements of the covariance matrix are parametrized. If left as the default `False`,
+        then :class:`~tfp.layers.MultivariateNormalTriL` layer is used where correlations
+        between outputs are learned as well.
+    :return: Keras ensemble model.
+    """
+    input_tensor_spec, output_tensor_spec = get_tensor_spec_from_data(dataset)
+
+    hidden_layer_args = []
+    for i in range(num_hidden_layers):
+        hidden_layer_args.append({"units": units, "activation": activation})
+
+    networks = [
+        GaussianNetwork(
+            input_tensor_spec,
+            output_tensor_spec,
+            hidden_layer_args,
+            independent_normal,
+        )
+        for _ in range(ensemble_size)
+    ]
+    keras_ensemble = KerasEnsemble(networks)
+
+    return keras_ensemble
+
+
+class KerasEnsemble(tf.Module):
+    """
+    This class builds an ensemble of neural networks, using Keras. Individual networks must
+    be instance of :class:`~trieste.models.keras_networks.KerasEnsembleNetwork`. This class
+    is meant to be used with :class:`~trieste.models.keras_networks.DeepEnsemble` model wrapper,
+    which compiles the model.
+    """
+
+    def __init__(
+        self,
+        networks: List[KerasEnsembleNetwork],
+    ) -> None:
+        """
+        :param networks: A list of neural network specifications, one for each member of the
+            ensemble. The ensemble will be built using these specifications.
+        """
+
+        super().__init__()
+
+        for network in networks:
+            if not isinstance(network, KerasEnsembleNetwork):
+                raise ValueError(
+                    f"Individual networks must be an instance of KerasEnsembleNetwork, "
+                    f"received {type(network)} instead."
+                )
+
+        self._networks = networks
+        self._model = self.build_ensemble()
+
+    def __repr__(self) -> str:
+        """"""
+        return f"KerasEnsemble({self._networks!r})"
+
+    @property
+    def model(self) -> tf.keras.Model:
+        """ " Returns built but uncompiled Keras ensemble model."""
+        return self._model
+
+    @property
+    def ensemble_size(self) -> int:
+        """
+        Returns the size of the ensemble, that is, the number of base learners or individual neural
+        network models in the ensemble.
+        """
+        return len(self._networks)
+
+    def build_ensemble(self) -> tf.keras.Model:
```

Comment:
Are people expected to ever call this again after construction? If not, clearer if you make the method private.

```suggestion
    def _build_ensemble(self) -> tf.keras.Model:
```